apps/prairielearn/src/schemas/schemas/infoAssessment.ts

```diff
@@ -0,0 +1,373 @@
+      .boolean()
+      .describe(
+        'Whether the student can view the assessment grade after it has been closed. Only works if showClosedAssessment is also set to false',
+      )
+      .optional(),
+    active: z
+      .boolean()
+      .describe(
+        'Whether the student can create a new assessment instance and submit answers to questions. If set to false, the available credit must be 0.',
+      )
+      .optional(),
+  })
+  .strict()
+  .describe(
+    'An access rule that permits people to access this assessment. All restrictions in the rule must be satisfied for the rule to allow access.',
+  );
+
+export type AsssessmentAccessRule = z.infer<typeof AsssessmentAccessRuleSchema>;
+
+export const PointsSingleSchema = z.number().gte(0).describe('A single point value.');
+
+export const PointsListSchema = z
+  .array(PointsSingleSchema)
+  .min(1)
+  .describe('An array of point values.');
+
+export const PointsSchema = z.union([PointsSingleSchema, PointsListSchema]);
+
+export const QuestionIdSchema = z
+  .string()
+  .describe('Question ID (directory name of the question).');
+
+export const ForceMaxPointsSchema = z
+  .boolean()
+  .describe('Whether to force this question to be awarded maximum points on a regrade.');
+
+export const AdvanceScorePercSchema = z
+  .number()
+  .gte(0)
+  .lte(100)
+  .describe('Minimum score percentage to unlock access to subsequent questions');
+
+const QuestionAlternativeSchema = z.object({
+  comment: z
+    .union([z.string(), z.array(z.any()), z.object({}).catchall(z.any())])
+    .describe('Arbitrary comment for reference purposes.')
+    .optional(),
+  points: PointsSchema.optional(),
+  autoPoints: PointsSchema.optional(),
+  maxPoints: PointsSingleSchema.optional(),
+  maxAutoPoints: PointsSingleSchema.optional(),
+  manualPoints: PointsSingleSchema.optional(),
+  id: QuestionIdSchema, // .optional(),
+  forceMaxPoints: ForceMaxPointsSchema.optional(),
+  triesPerVariant: z
+    .number()
+    .describe('The maximum number of graded submissions allowed for each question instance.')
+    .optional(),
+  advanceScorePerc: AdvanceScorePercSchema.optional(),
+  gradeRateMinutes: z
+    .number()
+    .gte(0)
+    .describe(
+      'Minimum amount of time (in minutes) between graded submissions to the same question.',
+    )
+    .optional(),
+  // TODO: What are these?
+  canView: z.array(z.string()).nullable().optional(),
+  canSubmit: z.array(z.string()).nullable().optional(),
+});
+
+const ZoneQuestionSchema = z.object({
+  comment: z
+    .union([z.string(), z.array(z.any()), z.object({}).catchall(z.any())])
+    .describe('Arbitrary comment for reference purposes.')
+    .optional(),
+  points: PointsSchema, // .optional(),
+  autoPoints: PointsSchema, // .optional(),
+  maxPoints: PointsSingleSchema, // .optional(),
+  maxAutoPoints: PointsSingleSchema, // .optional(),
+  manualPoints: PointsSingleSchema, // .optional(),
+  id: QuestionIdSchema.optional(),
```

Comment:
The code here: https://github.com/PrairieLearn/PrairieLearn/blob/33d10056121c78138c735122c031fbe5a737688a/apps/prairielearn/src/sync/course-db.ts#L1197-L1203 requires these to exist. Otherwise, I will provide a fallback value to the code.

Reply:
Resolved in [786dc11](https://github.com/PrairieLearn/PrairieLearn/pull/11024/commits/786dc11976e96104c9b0dbc4f44cb5881508e61e) and [0c0da20](https://github.com/PrairieLearn/PrairieLearn/pull/11024/commits/0c0da209cc933535bfa02367e3b21d58798b69a3).